### DISCUSSION TITLE
Pathways for adopting FSE

### DISCUSSION THREAD

**Discussion by annezazu** [WordPress/developer-blog-content#17] [Topic Ideas]
I think writing a post or series of posts highlighting various gradual adoption pathways would go a long way to raise awareness. Within these posts, compared to documentation, it would be rad to share more in depth code examples and walk through some of the practicalities (for ex: using theme.json in classic theme and impact on layout https://gutenbergtimes.com/layouts-and-wide-alignments-in-wordpress-then-now-and-upcoming-changes/). 

Could use this as a base but then add in more details and talk about factors to consider when adopting: https://developer.wordpress.org/themes/block-themes/converting-a-classic-theme-to-a-block-theme/

**Comment by bph** [comment]
Thank you, @annezazu That's a great topic, and I think you are right about the post series part! 😸 

Another example: [Building a Block-Based Header Template in a Classic Theme](https://gutenbergtimes.com/building-a-block-based-header-template-in-a-classic-theme/)

I'm not quite sure about the angle of a blog post that wouldn't replicate the available documentation page.  How would you approach it? 
A post that walks the reader through a particular task like @justintadlock did with the Block-based Header, could really work, if we can come up with some good problems to solve, that are hard to solve with classic but where blocks can shine

**Comment by annezazu** [comment]
I think I'd approach it in a more practical and robust way, perhaps talking about shortcomings, things to consider when it comes to the different pathways, examples of themes/folks using the pathways, etc (with more code examples!). The documentation is inherently brief and descriptive but there's lots of room left to help folks have more understanding.

**Comment by justintadlock** [comment]
_Just some free-flowing thoughts after reading..._

There are a ton of ways to approach this, and this is really in my wheelhouse (it's the sort of thing I've been blogging about for a decade and a half).

There are some really big things, such as the overall concept of incorporating block-based template parts into classic themes.  The [building a block-based header](https://gutenbergtimes.com/building-a-block-based-header-template-in-a-classic-theme/) post required a ton of extra content because it is both introducing a concept and following through with a real-world example.

In an ideal world where the Developer Blog was already live, I would've written an introductory tutorial that covered all of the foundational concepts.  Then, I would follow that up with very specific tutorials that built off of that, such as incorporating a hero header, 404 page content, and some of the other ideas.  Then, just link those tutorials back to the original.

This is one of the primary reasons I'm excited about the Developer Blog.  It provides a way to build off our existing content without needing to reintroduce concepts every time a new post is out.

I could easily write dozens of tutorials on concepts like block patterns and styles using this approach.  But, others could also build on top of the foundation.

For content planning, I'd go with something akin to a series.  Write a few introductory posts to various concepts, such as:

- Using theme.json in classic themes
- Building block patterns
- Creating custom block styles

Once those introductory posts are in place, we'd open this up to allow contributors to add posts in each series.  So, under the "Building block patterns" series, you might have new posts come along:

- Creating a hero header pattern for single posts
- Building pricing tables with a simple pattern
- A consistent design experience with content-only pattern locking

Then, have a small bit in each post:

> This post is part of the "Building block patterns" series.  Please check out the "Introduction post" for a walk-through of the basics.

There are several of these areas around theming that could use this treatment.  And, it would make less work for contributing writers not having to reexplain foundational concepts with each post.

**Comment by annezazu** [comment]
>For content planning, I'd go with something akin to a series. Write a few introductory posts to various concepts, such as:

Love this. You seem like the perfect person to take this very high impact work on!

**Comment by bph** [comment]
Converted this to a discussion for broader input and to narrow down scope for concrete posts or post series.
